grommet/grommet
# [TS] Fix RadioButton, RadioButtonGroup, RangeInput and RangeSelector type interfaces

#### What does this PR do?

This PR adds and exports `<component>ExtendedProps` declarations for the RadioButton, RadioButtonGroup, RangeInput and RangeSelector components, forming part of issue #4998.

#### Where should the reviewer start?

The only files changed are the TS declaration files for each of the four components. Any of these files are good places to start.

#### What testing has been done on this PR?

No new tests were added, all tests passed locally on each commit's pre-commit hook.

#### How should this be manually tested?

Importing the extended prop interfaces from a separate project should allow `JSX.IntrinsicElements` keys to be passed through to the imported interface, e.g. importing `RadioButtonExtendedProps` should make `input` keys available, as well as general HTML keys like `children`. Also double check any keys included in `Omit<..., ...>` types for consistency with original intersected types.

#### Any background context you want to provide?

All background context is available in this discussion/design issue: #4978

#### What are the relevant issues?

Progress tracking: #4998
Design/discussion: #4978

#### Do the grommet docs need to be updated?

No

#### Should this PR be mentioned in the release notes?

Not this PR specifically, but users should be made aware of the new extended prop type declarations.

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible.
